### PR TITLE
Fix #1904 changing type to none when PUT attr value (text/plain)

### LINKED ISSUE #1904: [BUG] in PUT /v2/entities/<id>/attrs/<attr_name>/value the attr_type and metadata_type are replaced by "none"
in PUT `/v2/entities/<id>/attrs/<attr_name>/value` request,  the attribute type and metadata type  always are replaced by **"none"**.

#### Create request
```
POST http://localhost:1026/v2/entities
Content-Type: application/json
Fiware-Service: test_update_content_type_text
Fiware-ServicePath: /test
   {
             "temperature_2": { "type": "celsius",  "value": "34",  "metadata": { "very_hot_1": {"type": "alarm", "value": "hot"},  "very_hot_0": {"type": "alarm", "value": "hot"}}}, 
             "temperature_0": {
                    "type": "celsius", 
                    "value": "34", 
                    "metadata": {
                           "very_hot_1": {"type": "alarm", "value": "hot"}, 
                           "very_hot_0": {"type": "alarm", "value": "hot"}}}, 
              "temperature_1": { "type": "celsius", "value": "34", "metadata": { "very_hot_1": {"type": "alarm", "value": "hot"}, "very_hot_0": {"type": "alarm", "value": "hot"}}}, 
              "type": "house",
              "id": "room_1"
   }
```

#### Update request
```
PUT http://localhost:1026/v2/entities/room_1/attrs/temperature_0/value
Content-Type: text/plain
Fiware-Service: test_update_content_type_text
Fiware-ServicePath: /test
true
```

#### Update response
```
http code: 204
date: Mon, 14 Mar 2016 16:06:57 GMT
connection: Keep-Alive
content-length: 0
```

#### Mongo doc
```
     {  
          "_id" : { "id" : "room_1", "type" : "house", "servicePath" : "/test" }, 
          "attrNames" : [ "temperature_2", "temperature_0", "temperature_1" ], 
          "attrs" : {
                  "temperature_2" : { "type" : "celsius", "creDate" : 1457971617, "modDate" : 1457971617, "value" : "34", "md" : [ { "name" : "very_hot_1", "type" : "alarm", "value" : "hot" }, { "name" : "very_hot_0", "type" : "alarm", "value" : "hot" } ] }, 
                  "temperature_0" : { 
                         "value" : true, 
                         "type" : "none", 
                         "md" : [ 
                                    { "name" : "very_hot_1", "type" : "none", "value" : "hot" }, 
                                    { "name" : "very_hot_0", "type" : "none", "value" : "hot" } ], 
                         "creDate" : 1457971617, 
                         "modDate" : 1457971617 }, 
                 "temperature_1" : { "type" : "celsius", "creDate" : 1457971617, "modDate" : 1457971617, "value" : "34", "md" : [ { "name" : "very_hot_1", "type" : "alarm", "value" : "hot" }, { "name" : "very_hot_0", "type" : "alarm", "value" : "hot" } ] } }, "creDate" : 1457971617, "modDate" : 1457971617 }

```

### PR DESCRIPTION
**Pull request by crbrox**
Fixed #1904 

@kzangeli 
@fgalan

### CODE REVIEW

**Comment by fgalan** [comment]
LGTM (passing the ball to @kzangeli for final LGTM + merge + delete branch)

File: test/functionalTest/cases/1904_attr_value_type_none/1904_attr_value_type_none.test [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+"temperature_0": {
+       "type": "celsius",
+       "value": "34",
+       "metadata": {
+              "very_hot_1": {"type": "alarm", "value": "hot"},
+              "very_hot_0": {"type": "alarm", "value": "hot"}}},
+ "temperature_1": { "type": "celsius", "value": "34", "metadata": { "very_hot_1": {"type": "alarm", "value": "hot"}, "very_hot_0": {"type": "alarm", "value": "hot"}}},
+ "type": "house",
+ "id": "room_1"
+}
```
Comment (kzangeli) [inline test/functionalTest/cases/1904_attr_value_type_none/1904_attr_value_type_none.test]: Why such a long payload?
It makes it more difficult to understand the test.
If necessary for the test, then OK, of course, but ... is it?

File: test/functionalTest/cases/1904_attr_value_type_none/1904_attr_value_type_none.test [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB 0-255
+
+--SHELL--
+
+#
+# 01. POST /v2/entities creating room
+# 02. PUT /v2/entities/room/attrs/temperature/value (forbidden chars)
```
Comment (kzangeli) [inline test/functionalTest/cases/1904_attr_value_type_none/1904_attr_value_type_none.test]: forbidden chars ... that's a cut 'n paste error, right?
I don't see any forbidden chars in '02' and its 'header text' doesn't contain the string either.

File: test/functionalTest/cases/1904_attr_value_type_none/1904_attr_value_type_none.test [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+echo '================================================='
+payload='true'
+orionCurl --url /v2/entities/room_1/attrs/temperature_0/value --payload "$payload" --in text -X PUT
+echo
+echo
+
+
+echo '03. GET /v2/entities/room (type is not "none")'
+echo '=============================================='
+orionCurl --url /v2/entities/room_1
```
Comment (kzangeli) [inline test/functionalTest/cases/1904_attr_value_type_none/1904_attr_value_type_none.test]: echo "03. GET /v2/entities/room_1 ...

Same mistake room/room_1  in 02 not that I look at it ...

File: test/functionalTest/cases/1904_attr_value_type_none/1904_attr_value_type_none.test (hunk 1 above)
Comment (crbrox) [inline test/functionalTest/cases/1904_attr_value_type_none/1904_attr_value_type_none.test]: Not necessary ... It just the same used by QA ( @iariasleon ) , trying to make easier to check against the issue

Comment (kzangeli) [inline test/functionalTest/cases/1904_attr_value_type_none/1904_attr_value_type_none.test]: OK, that is also a very good reason for it :-)  

NTC

File: test/functionalTest/cases/1904_attr_value_type_none/1904_attr_value_type_none.test (hunk 2 above)
Comment (crbrox) [inline test/functionalTest/cases/1904_attr_value_type_none/1904_attr_value_type_none.test]: Sure, what a mess!
Znx

fixed at 5876c2a2fd0a9f6730f7ff53e6363bc656ef3f91

File: test/functionalTest/cases/1904_attr_value_type_none/1904_attr_value_type_none.test (hunk 3 above)
Comment (crbrox) [inline test/functionalTest/cases/1904_attr_value_type_none/1904_attr_value_type_none.test]: fixed at 5876c2a2fd0a9f6730f7ff53e6363bc656ef3f91

**Comment by kzangeli** [comment]
LGTM

**Pull request merged**